docs/timezone.md

```diff
@@ -127,6 +133,8 @@ tz2 = Temporal.TimeZone.from(tz);
 The `id` property gives an unambiguous identifier for the time zone.
 Effectively, this is the canonicalized version of whatever `timeZoneIdentifier` was passed as a parameter to the constructor.
 
+The default implementation of this property gives the result of calling `toString()`.
```

Comment:
Minor concern: will talking about the default implementations inside each method's docs be confusing for the vast majority of developers who will never implement a custom time zone? 

Most developer documentation separates the "plugin authors" part of the docs from the "plugin users" part of the docs because there are 100x+ more of the latter and including info that only the former will use may be confusing. So IMHO, instead of combining custom time zone info into the "users" page, it seems better to have a separate page that's dedicated to people creating custom zones or calendars. This would simplify the docs for most users while giving the relevant info for the minority of advanced users who need it. 

I'd volunteer to do this split in a separate PR if you think this would be worthwhile-- I know you're super-busy on lots of things that I can't help with, but I can help with docs! ;-)

Reply:
For the same reason I thought that the text you had earlier was too long, I don't think that we even want to extensively document the plain object time zones, definitely not on a separate page that makes them into A Thing that people might think they should do. Rather I'd consider them just a peculiarity of how protocols work in JS.

If this wording might be confusing, how about adding "When subclassing Temporal.TimeZone," to each of these sentences?

Reply:
OK I'm convinced. 



> If this wording might be confusing, how about adding "When subclassing Temporal.TimeZone," to each of these sentences?

This seems like a great solution. Thanks!

Reply:
> "When subclassing Temporal.TimeZone,"

It might be good to clarify that the methods with this message don't need to be overridden, for example: 
* "When subclassing Temporal.TimeZone, this method does not need to be overridden because the default implementation..."
* (for the getXxxTransition methods) "When subclassing Temporal.TimeZone, this method should be overridden if the time zone changes offsets. Single-offset time zones can use the default implementation which returns `null`."